eng/pipelines/aggregate-reports.yml

```diff
@@ -17,144 +27,179 @@ pr:
+          - task: PowerShell@2
+            displayName: 'Validate dependencies with pwsh stable'
+            inputs:
+              pwsh: true
+              filePath: 'eng/scripts/dependencies/compare-deps-files.ps1'
+              arguments: >
+                -PSDepsFile 'https://aka.ms/ps-deps-stable'
+                -AzSdkDepsFile '$(Build.ArtifactStagingDirectory)/reports/azure-sdk.deps.json'
+          - task: PowerShell@2
+            displayName: 'Validate dependencies with pwsh preview'
+            inputs:
+              pwsh: true
+              filePath: 'eng/scripts/dependencies/compare-deps-files.ps1'
+              arguments: >
+                -PSDepsFile 'https://aka.ms/ps-deps-preview'
+                -AzSdkDepsFile '$(Build.ArtifactStagingDirectory)/reports/azure-sdk.deps.json'
+          - task: PublishPipelineArtifact@1
+            displayName: 'Publish Report Artifacts'
+            inputs:
+              artifactName: reports
+              path: '$(Build.ArtifactStagingDirectory)/reports'
+          - task: AzureFileCopy@2
+            displayName: 'Upload dependency report'
+            inputs:
+              sourcePath: '$(Build.ArtifactStagingDirectory)/reports'
+              azureSubscription: 'Azure SDK Artifacts'
+              destination: AzureBlob
+              storage: azuresdkartifacts
+              containerName: 'azure-sdk-for-net'
+              blobPrefix: dependencies
+          - task: PowerShell@2
+            displayName: "Verify Repository Resource Refs"
+            inputs:
+              pwsh: true
+              workingDirectory: $(Build.SourcesDirectory)
+              filePath: eng/common/scripts/Verify-Resource-Ref.ps1
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
+            displayName: 'Run CredScan'
+            inputs:
+              suppressionsFile: 'eng\CredScanSuppression.json'
+            condition: succeededOrFailed()
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+            displayName: 'Post Analysis'
+            inputs:
+              GdnBreakAllTools: false
+              GdnBreakGdnToolCredScan: true
+              GdnBreakGdnToolCredScanSeverity: Error
+              GdnBreakBaselineFiles: $(Build.SourcesDirectory)\eng\dotnet.gdnbaselines
+              GdnBreakBaselines: baseline  
+              # Used for generating baseline file
+              # GdnBreakOutputBaselineFile: dotnet
+              # GdnBreakOutputBaseline: baseline
+            condition: succeededOrFailed()
+            continueOnError: true
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
+            displayName: 'Publish Security Analysis Logs'
+            condition: succeededOrFailed()
+
+  - stage: GuardianTools
```

Comment:
To make this a little more generic perhaps we just call this compliance like we did in java. https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/aggregate-reports.yml#L111